src/app-layer-dhcp.c

```diff
@@ -0,0 +1,699 @@
+                    case DHCP_REQUEST:
+                        tx = DHCPGetTxByXid(dhcp_state, bootp->xid);
+                        if (unlikely(tx == NULL)) {
+                            SCLogDebug("Failed to allocate new DHCP tx.");
+                            goto end;
+                        }
+                        tx->request_buffer_len = input_len - sizeof(BOOTPHdr);
+                        tx->request_buffer = SCMalloc(tx->request_buffer_len);
+                        if (unlikely(tx->request_buffer == NULL)) {
+                            /* TBD: need to remove from global tx list */
+                            DHCPTxFree(global, tx, 0);
+                            goto end;
+                        }
+                        memcpy(tx->request_buffer, dhcp, tx->request_buffer_len);
+                        tx->request_seen = 1;
+                        break;
+                    case DHCP_INFORM:
+                        tx = DHCPGetTxByXid(dhcp_state, bootp->xid);
+                        if (unlikely(tx == NULL)) {
+                            SCLogDebug("Failed to allocate new DHCP tx.");
+                            goto end;
+                        }
+                        tx->request_client_ip = bootp->ciaddr;
+                        tx->request_buffer_len = input_len - sizeof(BOOTPHdr);
+                        tx->request_buffer = SCMalloc(tx->request_buffer_len);
+                        if (unlikely(tx->request_buffer == NULL)) {
+                            /* TBD: need to remove from global tx list */
+                            DHCPTxFree(global, tx, 0);
+                            goto end;
+                        }
+                        memcpy(tx->request_buffer, dhcp, tx->request_buffer_len);
+                        tx->request_seen = 1;
+                        break;
+                    case DHCP_RELEASE:
+                    case DHCP_DECLINE:
+                        tx = DHCPGetTxByXid(dhcp_state, bootp->xid);
+                        if (unlikely(tx == NULL)) {
+                            SCLogDebug("Failed to allocate new DHCP tx.");
+                            goto end;
+                        }
+                        tx->request_buffer_len = input_len - sizeof(BOOTPHdr);
+                        tx->request_buffer = SCMalloc(tx->request_buffer_len);
+                        if (unlikely(tx->request_buffer == NULL)) {
+                            /* TBD: need to remove from global tx list */
+                            DHCPTxFree(global, tx, 0);
+                            goto end;
+                        }
+                        memcpy(tx->request_buffer, dhcp, tx->request_buffer_len);
+                        /* response to release not required */
+                        tx->response_unneeded = 1;
+                        tx->request_seen = 1;
+                        break;
+                    default:
+                        SCLogDebug("DHCP unknown %d", dhcp->args[0]);
```

Comment:
I'm not sure if it should, yet. This more of a we don't handle this message yet, and not an uknown. When/if we exhaustively handle all known message types then maybe it should be alert? But the scope of this is more about logging messages to track allocated addresses.